scb-persistenceproviders/defectdojo-persistenceprovider/src/main/java/io/securecodebox/persistence/DefectDojoService.java

```diff
@@ -314,21 +442,24 @@ public void deleteUnusedBranches(List<String> existingBranches, String producNam
      * Be aware that the branch tag MUST be set, otherwise all engagments will be deleted
      */
     public void deleteUnusedBranches(List<String> existingBranches, long productId) {
+        if(existingBranches == null) {
+            LOG.error("No existing branches given, this will lead to nullpointer");
+        }
         RestTemplate restTemplate = new RestTemplate();
         
         //get existing branches
         List<EngagementResponse> engagementPayloads = getEngagementsForProduct(productId, 0);
         for(EngagementResponse engagementPayload : engagementPayloads) {
-            boolean branchExists = false;
+            boolean branchExists = false
```

Comment:
Missing semicolon is breaking the build.

```suggestion
            boolean branchExists = false;
```